include/boost/geometry/io/wkt/write.hpp

```diff
@@ -62,22 +63,67 @@ namespace boost { namespace geometry
 namespace detail { namespace wkt
 {
 
+template <typename Output>
+struct output_formatter
+{
+    template <typename T>
+    static void append(Output& out, T const& v)
+    {
+        out << v;
+    }
+};
+
+template <typename CharT, typename Traits, typename Allocator>
+struct output_formatter<std::basic_string<CharT, Traits, Allocator>>
+{
+    template <typename T>
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, T const& val)
+    {
+        std::basic_stringstream<CharT, Traits, Allocator> ss;
+        ss << val;
+        out += ss.str();
+    }
+
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, CharT c)
+    {
+        out += c;
+    }
+
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, const CharT* s)
```

Comment:
As @mloskot said earlier the type should be `CharT const*`

Reply:
The problem with this overload and the one above is that they take `CharT` and `CharT const*` which are taken from `std::basic_string`. So if this type is anything else then the first overload of `append()` taking any `T const&` is going to be instantiated. This will happen e.g. when you pass `std::wstring` because this type is derived from `std::basic_string<wchar_t>` [1] so `CharT` will be `wchar_t` but below values of type `char` or `char const*` are passed into `append()`, e.g. `' '` or `"("`.

This is not a problem now, because `to_wkt` always returns `std::string` but if you wanted it to be complete, and e.g. allowed to define the output string type, e.g. like that:

std::wstring wkt = bg::to_wkt<std::wstring>(geom);

then this implementation would be sub-optimal because for all chars and string literals `std::basic_stringstream` would be used anyway.

[1] https://en.cppreference.com/w/cpp/string/basic_string

Reply:
I agree with @awulkiew here. I missed the fact all the current WKT IO implementation, by design or not, quite assumes `char`-based output.

Adding support for arbitrary `CharT` is something that could be considered but in separate PR.

Then, with help of [additional layer of indirection](http://www.josuttis.com/tmplbook/), we could specify required characters and WKT tags too using expected types, somethign along these lines:

```
template <typename CharT>
struct output_char_traits;
template <>
struct output_char_traits<char> { static const char space = ' '; };
template <>
struct output_char_traits<wchar_t> { static const wchar_t space = L' '; };

template <typename Output>
void apply(Output& out)
{
    // traits_type common for basic_string and basic_stringstream
    typedef typename Output::traits_type::char_type char_type;
    output_formatter<Output>::append(out, output_char_traits<char_type>::space);
};
```

Reply:
It is also possible that this could be simplified. I didn't look into it but it's possible that the codes of the basic characters we use (' ', '(', ')', ',', etc.) are the same for all character types (`char`, `wchar_t`, `char16_t`, `char32_t`, etc.). So it's possible that a simple `static_cast` would be enough.

Reply:
The cast may not work in a endian-agnostic manner. There is however narrow/widen functions defined and the standard guarantees unambiguous transformation for the [basic charactes  set of 96 characters](https://books.google.pl/books?id=5qybChHSk7QC&pg=PA279&lpg=PA279&dq=char+widen+narrow&source=bl&ots=drhQ3F5n20&sig=ACfU3U0bF4Ro_vwSXVE56F9TGIytpMrWPQ&hl=en&sa=X&ved=2ahUKEwjWiajLsP_nAhXksIsKHd3_DocQ6AEwA3oECAYQAQ#v=onepage&q=char%20widen%20narrow&f=false)

Reply:
As you have said that the support for CharT should be added in a separate PR, what is to be done in this PR?

Reply:
@sudo-panda As you see the review turned into some additional brainstorm, not uncommon. 

I think reverting back to explicit use of narrow character `std::string` and `std::stringstream` is probably the right thing to do, but let's wait for @awulkiew opinion on that.

Extending current all the plain text formatters (WKT and even SVG) to support arbitrary type of character is something desirable, I think, but it is a task separate from this PR.

Reply:
> I think reverting back to explicit use of narrow character std::string and std::stringstream is probably the right thing to do
@mloskot Yes, I'm ok with that.